adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m

```diff
@@ -0,0 +1,86 @@
+//
+//  GADPangleRTBBannerRenderer.m
+//  PangleAdapter
+//
+//  Created by bytedance on 2022/1/11.
+//
+
+#import "GADPangleRTBBannerRenderer.h"
+#import "GADMediationAdapterPangleConstants.h"
+#import <BUAdSDK/BUAdSDK.h>
+#import "GADMAdapterPangleUtils.h"
+
+@interface GADPangleRTBBannerRenderer()<BUNativeExpressBannerViewDelegate>
+
+@end
+
+@implementation GADPangleRTBBannerRenderer {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationBannerLoadCompletionHandler _loadCompletionHandler;
+    /// The Pangle banner ad.
+    BUNativeExpressBannerView *_nativeExpressBannerView;
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationBannerAdEventDelegate> _delegate;
+    /// The requested ad size.
+    CGSize _bannerSize;
+}
+
+- (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
+                     completionHandler:
+(nonnull GADMediationBannerLoadCompletionHandler)completionHandler {
+    _loadCompletionHandler = completionHandler;
+    NSString *slotId = adConfiguration.credentials.settings[GADMAdapterPanglePlacementID] ?: @"";
+    if (PangleIsEmptyString(slotId)) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorSlotIdNil, [NSString stringWithFormat:@"%@ cannot be nil.",GADMAdapterPanglePlacementID]);
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    _bannerSize = adConfiguration.adSize.size;
+    _nativeExpressBannerView = [[BUNativeExpressBannerView alloc]initWithSlotID:slotId rootViewController:adConfiguration.topViewController adSize:adConfiguration.adSize.size];
+    _nativeExpressBannerView.delegate = self;
+    if (![_nativeExpressBannerView respondsToSelector:@selector(setAdMarkup:)]) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorVersionLow, @"Pangle SDK version is too low");
```

Comment:
Question, why is this error message "Pangle SDK version is too low"?

Reply:
if ad object doesn't support this selector  because the version is too low , publishers should update their's Pangle SDK

Reply:
Can you remind me again which Pangle SDK (and its version) should be used with this adapter?

Does the publishers need to import the SDK separately? The release of this Pangle iOS adapter should already include it as the dependency.

Reply:
Mark as unresolved.

Reply:
Pangle SDK version >= 4.3.0.0